### ISSUE TITLE
You have found a bug in locals

### ISSUE THREAD

**Issue by mrjbq7** [slavapestov/factor#1341]
Okay, maybe related to #1338 and #1340, but this is weird.

First, paste this definition, fixing the restarts.

```factor
:: sieve1 ( n -- primes )
    n 1 + t <array> :> sieve
    f 0 sieve set-nth
    f 1 sieve set-nth

    2 n sqrt [a,b] [| i |
        i sieve nth [
            i sq n i <range> [| j |
                f j sieve set-nth
            ] each
        ] when
    ] each

    sieve [ drop ] selector [ each-index ] dip ;
```

Next, paste this definition, fixing the restarts.

```factor
:: sieve2 ( n -- primes )
    n 1 + <bit-array> :> sieve
    t 0 sieve set-nth
    t 1 sieve set-nth

    2 n sqrt [a,b] [| i |
        i sieve nth [
            i sq n i <range> [| j |
                t j sieve set-nth
            ] each
        ] unless
    ] each

    sieve [ drop not ] selector [ each-index ] dip ;
```

After fixing the restart for ``bit-arrays``, you get this:

```
You have found a bug in locals. Please report.
args [ ]
obj  sieve
```

**Comment by mrjbq7** [comment]
I don't remember so many issues with locals, I wonder if this is somehow related to the recent compiler changes.

**Comment by bjourne** [comment]
Weird. It must have something to do with the restarts because if you compile it with the right USING:s it works. Same goes for the other bugs.

**Comment by mrjbq7** [comment]
Seems to be fixed with #1340!
